### PR TITLE
Us105 address search pagination

### PR DESCRIPTION
**Pull request by nick-gravgaard**

### CODE REVIEW

File: service/server.py [hunk 1]
```diff
@@ -197,68 +175,57 @@ def find_titles(search_term=''):
...
         postcode = sanitise_postcode(search_term)
-        postcode_search_results = get_register_titles_via_postcode(postcode)
-        return render_search_results(postcode_search_results, postcode)
-    elif search_term:
-        address_search_results = get_register_titles_via_address(search_term)
-        return render_search_results(address_search_results, search_term)
-
+        postcode_search_results = get_register_titles_via_postcode(postcode, page_num)
+        return render_search_results(postcode_search_results, postcode, page_num)
+    else:
```
Comment (lgonczar) [inline service/server.py:202]: Recently Camille made a change to this method, that would prevent search by address when the search term is empty. Your version doesn't do it anymore. Is it deliberate?

File: service/server.py [hunk 2]
```diff
@@ -326,29 +287,27 @@ def _is_invalid_credentials_response(response):
...
+    password = PasswordField('password', [Required(message='Password is required')])
 
     def __init__(self, *args, **kwargs):
         Form.__init__(self, *args, **kwargs)
 
 
 class TitleSearchForm(Form):
-    # Form used for providing CSRF tokens for title search HTTP form
-    pass
+    search_term = StringField('username', [Required(message='Username is required'),
```
Comment (lgonczar) [inline service/server.py]: The search term field is named `username` here

File: service/server.py [hunk 3]
```diff
@@ -326,29 +287,27 @@ def _is_invalid_credentials_response(response):
...
+                                           Length(min=4, max=70, message='Username is incorrect')])
 
+    def __init__(self, *args, **kwargs):
+        Form.__init__(self, *args, **kwargs)
 
-if _is_csrf_enabled():
-    CsrfProtect(app)
+
+def _is_csrf_enabled():
+    return app.config.get('DISABLE_CSRF_PREVENTION') is not True
```
Comment (lgonczar) [inline service/server.py:308]: :+1:

File: service/templates/search_results.html [hunk 4]
```diff
@@ -15,38 +15,61 @@
...
-                        </h1>
-                        <input type="text" class="form-control" name="search_term" id="search_term" value="{% if search_term %}{{ search_term }}{% endif %}">
-                        <input type="submit" value="Search" class="button search-button">
-                      </div>
+                    <form class="form" action="{{ url_for('find_titles') }}" method="post" role="form">
+                        {{ form.hidden_tag() }}
+                        <div class="form-group">
+                            <h1 class="heading-medium">
+                                <label for="search_term">
+                                Search results for <span class="visuallyhidden">{% if search_term %}{{ search_term }}{% endif %}</span>
```
Comment (lgonczar) [inline service/templates/search_results.html:23]: If we never search by an empty term, we can get rid of this `if`

File: tests/test_app.py [hunk 5]
```diff
@@ -186,6 +192,18 @@ def test_postcode_bad_space_search_success(self):
         postcode = sanitise_postcode(search_term)
         assert postcode == 'PL9 8TB'
 
+    @mock.patch('requests.get', return_value=fake_postcode_search)
+    def test_postcode_search_with_page_calls_api_correctly(self, mock_get):
+        self.app.get('/title-search/PL9%207FN?page=23')
+        mock_get.assert_called_with('http://landregistry.local:8004/title_search_postcode/PL9 7FN',
+                                    params={'page': 23})
+
+    @mock.patch('requests.get', return_value=fake_address_search)
```
Comment (lgonczar) [inline tests/test_app.py:201]: These tests are good. I've also been thinking about these cases: page being less then 1, not a number, greater than the number of pages (?). Do we have tests that check if we present what the API gave us?

**Comment by lgonczar** [comment]
Reviewed. Lots of good stuff. Left some comments. It would be good for one more person to have a look.

**Comment by camillebaldock** [comment]
LGTM

**Pull request merged**